klayout/drc/rule_decks/main.drc

```diff
@@ -0,0 +1,1071 @@
+################################################################################################
+# Copyright 2022 GlobalFoundries PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################################
+
+#===========================================================================================================================
+#------------------------------------------- GF 0.18um MCU DRC RULE DECK --------------------------------------------------
+#===========================================================================================================================
+require 'time'
+require "logger"
+require 'etc'
+
+exec_start_time = Time.now
+
+logger = Logger.new(STDOUT)
+
+logger.formatter = proc do |severity, datetime, progname, msg|
+  "#{datetime}: Memory Usage (" + `pmap #{Process.pid} | tail -1`[10,40].strip + ") : #{msg}
+"
+end
+
+#================================================
+#----------------- FILE SETUP -------------------
+#================================================
+
+# optional for a batch launch :   klayout -b -r gf_018mcu.drc -rd input=design.gds -rd report=gp180_drc.lyrdb
+
+logger.info("Starting running GF180MCU Klayout DRC runset on %s" % [$input])
+logger.info("Ruby Version for klayout: %s" % [RUBY_VERSION])
+
+if $input
+    if $topcell
+        source($input, $topcell)
+    else
+        source($input)
+    end
+end
+
+if $table_name
+  table_name = $table_name
+else
+  table_name = "main"
+end
+
+
+logger.info("Loading database to memory is complete.")
+
+if $report
+    logger.info("GF180MCU Klayout DRC runset output at: %s" % [$report])
+    report("DRC Run Report at", $report)
+else
+    logger.info("GF180MCU Klayout DRC runset output at default location." % [File.join(File.dirname(RBA::CellView::active.filename), "gf180_drc.lyrdb")])
+    report("DRC Run Report at", File.join(File.dirname(RBA::CellView::active.filename), "gf180_drc.lyrdb"))
```

Comment:
I think you might want to save `File.join(File.dirname(RBA::CellView::active.filename), "gf180_drc.lyrdb")` to an intermediate variable, since it's being used in multiple places in that file.

Reply:
something like:
```
    layout_dir = Pathname.new(RBA::CellView::active.filename).parent.realpath
    report_path = layout_dir.join("gf180_drc.lyrdb").to_s
    logger.info("GF180MCU Klayout DRC runset output at default location: %s" % report_path)
    report("DRC Run Report at", report_path)
```
to ensure that you resolve the path appropriately.